src/services/State.ts

```diff
@@ -76,24 +77,21 @@ export class StateService {
     return { first, last };
   }
 
-  setSelected(selected: any[]): StateService {
-    if (!this.selected) {
-      this.selected = selected || [];
-    } else {
-      this.selected.splice(0, this.selected.length);
-      this.selected.push(...selected);
-    }
+  get selected(): any[] {
+    return this.rows.filter(row => this.isRowSelected(row));
```

Comment:
Well, in case we'd have like 100.000 rows, yeah, that could be slow. But that's not a problem, we could refactor setSelected and setRows to cache the `selected` value. I'll remake it.

`setSelected` - I'd leave it as it is, just like `setRows` and `setOptions`. Personally, I use getters/setters only if they are very simple (set/get value), but don't have any extra behaviour (like emitting events or sth).